### DISCUSSION TITLE
How to gather compled file to a folder I specify with tasks.json?

### DISCUSSION THREAD

**Discussion by Latzl** [microsoft/vscode-cpptools#9535] [Q&A]
I'm a beginner of C++ and I have a problem with configuration path.
It can work with sentence following in "arg:" 
"/Fe${fileDirname}\\${fileBasenameNoExtension}\\${fileBasenameNoExtension}.exe",
When I insert an **nonexistent folder** like following: 
"/Fe${fileDirname}\\**compiledFile**\\${fileBasenameNoExtension}\\${fileBasenameNoExtension}.exe",
and run a program with extension, the terminal just show starting generation and didn't complie any file.

These are my tasks.json, launch.json etc.
![task](https://user-images.githubusercontent.com/99496037/177034584-e1f9c48d-2569-4b06-9dea-78995ff2faad.png)

![cpp](https://user-images.githubusercontent.com/99496037/177034593-13ec5d8a-4d6a-499e-8d4b-275e15fae22c.png)

![launch](https://user-images.githubusercontent.com/99496037/177034596-293d6248-a591-44f6-a6e3-ce3058482d16.png)

Thanks so much!

**Comment by sean-mcmanus** [comment]
I'll look into this tomorrow...

**Reply by sean-mcmanus** [reply]
You'll need to change the "program" property in launch.json to match the path used for the "/Fe" flag, i.e. added compiledFile. If that still doesn't work, can you provide more detailed info on the error message seen, because I don't see to repro any other issue after that is resolved. You may also want to try 1.11.0 (pre-release).

**Reply by Latzl** [reply]
Thanks for replying.
The main problem is that it can't complie any file on a nonexistent folder and startint build permanently, show at p1.
![1](https://user-images.githubusercontent.com/99496037/177686031-99933c84-fd7c-45f4-9995-cd5614c07637.png)

If I create the folder, it can work normally show at p2 p3.
![2](https://user-images.githubusercontent.com/99496037/177686040-e597a3a9-a517-4aa3-91fe-787a7a32bb90.png)
![3](https://user-images.githubusercontent.com/99496037/177686050-95e8f8cb-254e-4ea0-bcc4-66d3cef94d26.png)

However, it still error when I specify a nonexistent subfolder in an existent folder, show at p4
![4](https://user-images.githubusercontent.com/99496037/177686062-6c4e0b61-a88e-4dc2-be85-ec5ead261ba2.png)

So， I think it's wrong to mix  variable substitution whith nonexistent path or there's something I didn't aware?

**Reply by sean-mcmanus** [reply]
Yeah, you can't use a non-existent path for those command line arguments, i.e. the compiler won't create those paths for you.

**Comment by Latzl** [comment]
tks, it was resolved by creating an specified folder with shell when each time run the script each time.
